ambari-server/src/main/java/org/apache/ambari/server/controller/internal/BlueprintConfigurationProcessor.java

```diff
@@ -499,6 +499,27 @@ private void doNameNodeHAUpdateOnClusterCreation(Configuration clusterConfig,
         if (parsedNameServices.length > 1) {
           Set<String> activeNameNodeHostnames = new HashSet<>();
           Set<String> standbyNameNodeHostnames = new HashSet<>();
+          String tagsyncNameserviceMappingProperty   = "ranger.tagsync.atlas.hdfs.instance." + getClusterName() + ".%s.ranger.service";
```

Comment:
Will this configuration support be required for traditional HDFS NameNode HA as well, meaning a single HDFS nameservice defined?  If so, then this code to generate the properties for Ranger should probably also execute in the above block of code.  If this property generation should only occur for Federated deployments, then this block is in the right place.

Reply:
@rnettleton, The properties will be required only for federation scenarios.

The service-advisor case will not be feasible here as we require the cluster-name for configuring the required properties. When the blueprint is deployed, the cluster-name will not be available in service-advisor to configure the required properties, so the deployment might fail.